services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3CrtTransferProgressListenerTest.java

```diff
@@ -160,7 +160,9 @@ void listeners_reports_ErrorsWhenCancelled(WireMockRuntimeInfo wm) throws Interr
         assertThat(transferListener.getExceptionCaught()).isInstanceOf(CancellationException.class);
         assertThat(transferListener.isTransferComplete()).isFalse();
         assertThat(transferListener.isTransferInitiated()).isTrue();
-        assertMockOnFailure(transferListenerMock);
+        Mockito.verify(transferListenerMock, times(1)).transferFailed(ArgumentMatchers.any());
```

Comment:
The failure of the test actually verified this fix. Before the fix, the request would never cancelled, so we'd still get data after cancelling future. After the fix, the request would get cancelled properly, so ` Mockito.verify(transferListenerMock, times(1)).bytesTransferred(ArgumentMatchers.any());` may not be invoked at all.